src/react/hooks/use-typing.ts

```diff
@@ -88,20 +88,13 @@ export const useTyping = (params?: TypingParams): UseTypingResponse => {
       return new Set<string>();
     });
 
-    let mounted = true;
-
     void context.room
       .then((room) => {
-        if (!mounted) return;
-
         // If we're not attached, we can't call typing.get() right now
         if (room.status === RoomStatus.Attached) {
           const typing = room.typing.get();
           logger.debug('useTyping(); room attached, getting initial typers', { typing });
           setCurrentlyTyping(typing);
-        } else {
-          logger.debug('useTyping(); room not attached, setting currentlyTyping to empty');
-          setCurrentlyTyping(new Set());
         }
       })
       .catch();
```

Comment:
I think these should go inside the wrapRoomPromise below instead of doing their own waiting? So the `unmount()` callback works correctly for those too?

It's likely this code isn't doing much though because at the time of mounting (at least on page load) the room very likely isn't yet attached. This block would make more sense in an effect that depends on room status.

It can be fixed in another PR, don't mean to drag this along. Maybe a ticket so we get back to it?

cc @splindsay-92 because it's for typing indicators.

Reply:
Yes, I agree - I've moved it into the same block.